### ISSUE TITLE
MD034 false positive for reference definitions with an escaped closing bracket in link labels

### ISSUE THREAD

**Issue by gifnksm** [rvben/rumdl#814]
### rumdl version

0.2.54

### Installation method

mise

### Bug description

MD034 (URL without angle brackets) is incorrectly reported for a reference definition with a link label containing an escaped closing bracket.

CommonMark spec says:

> A [link label](https://spec.commonmark.org/0.31.2/#link-label) begins with a left bracket ([) and ends with the first right bracket (]) that is not backslash-escaped. Between these brackets there must be at least one character that is not a space, tab, or line ending. Unescaped square bracket characters are not allowed inside the opening and closing square brackets of [link labels](https://spec.commonmark.org/0.31.2/#link-label). A link label can have at most 999 characters inside the square brackets.

(<https://spec.commonmark.org/0.31.2/#link-label>)

Escaped square brackets are allowed inside link labels. In particular, the CommonMark spec includes an example with an escaped opening bracket ([Example 549]). However, a link label containing an escaped closing bracket appears to be handled incorrectly by rumdl.

[Example 549]: https://spec.commonmark.org/0.31.2/#example-549

### Steps to reproduce

1. Create a file `test.md` with this content:

   ```markdown
   # Reference Example

   * [this is a link to example.com][ref1\[\]]
   * [this is also a link to example.com][ref2\[]
   * [this is also a link to example.com][ref3]

   [ref1\[\]]: https://example.com/ref1
   [ref2\[]: https://example.com/ref2
   [ref3]: https://example.com/ref3
   ```

2. Run `rumdl check test.md`
3. See the output message

### Expected behavior

No warnings are emitted

### Actual behavior

MD034 is incorrectly reported for the first reference definition.

```
test.md:7:13: [MD034] URL without angle brackets or link formatting: 'https://example.com/ref1' [*]

Issues: Found 1 issues in 1 file (8ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

### Configuration

```toml
No configuration
```

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in v0.2.56 (f024e4df). MD034's reference-definition guard reads the label the way CommonMark does, so a backslash-escaped `]` inside the label no longer ends it and the URL in `[a\]b]: https://...` is not reported as bare. The reversed direction (an unterminated label) is pinned by a test so the guard does not over-reach. Thanks for the spec pointer.
